Automattic/jetpack
# Social: Register new CPT for Social Posts

## Proposed changes:

* Added a new class `Jetpack_Social_Note` to register the custom post type
* I have added this class to the src folder of the plugin, but if need be it can be refactored into its own package in the future. 
* I have used `dashicons-welcome-write-blog` as the menu icon for now, we can tweak this as well if need be.


### Other information:


## Jetpack product discussion


pdrWKz-156-p2#comment-1932
## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Add `define( 'JETPACK_SOCIAL_NOTES_ENABLED', true );` to your `wp-config.php`. If you're on JN, install the code snippets plugin and copy `define( 'JETPACK_SOCIAL_NOTES_ENABLED', true );` into it. 
* Activate Jetpack Social on this branch
* Ensure you're able to see the new CPT Social Note in the sidebar of your wp-admin and you're able to create new Social Notes. 
* Set `JETPACK_SOCIAL_NOTES_ENABLED` to false and ensure that you no longer see the new CPT in the sidebar.
<img width="148" alt="Screenshot 2024-02-02 at 5 51 09 PM" src="https://github.com/Automattic/jetpack/assets/6594561/1a28712d-222b-4c4e-b357-6b39891ebe9d">
